pkg/tasks/prometheus_user_workload.go

```diff
@@ -0,0 +1,281 @@
+// Copyright 2019 The Cluster Monitoring Operator Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tasks
+
+import (
+	"github.com/openshift/cluster-monitoring-operator/pkg/client"
+	"github.com/openshift/cluster-monitoring-operator/pkg/manifests"
+
+	"github.com/pkg/errors"
+	"k8s.io/klog"
+)
+
+type PrometheusUserWorkloadTask struct {
+	client             *client.Client
+	factory            *manifests.Factory
+	userWorkloadConfig *manifests.UserWorkloadConfig
+}
+
+func NewPrometheusUserWorkloadTask(client *client.Client, factory *manifests.Factory, userWorkloadConfig *manifests.UserWorkloadConfig) *PrometheusUserWorkloadTask {
+	return &PrometheusUserWorkloadTask{
+		client:             client,
+		factory:            factory,
+		userWorkloadConfig: userWorkloadConfig,
+	}
+}
+
+func (t *PrometheusUserWorkloadTask) Run() error {
+	if t.userWorkloadConfig.IsEnabled() {
+		return t.create()
+	}
+
+	return t.destroy()
+}
+
+func (t *PrometheusUserWorkloadTask) create() error {
+	cacm, err := t.factory.PrometheusUserWorkloadServingCertsCABundle()
+	if err != nil {
+		return errors.Wrap(err, "initializing serving certs CA Bundle ConfigMap failed")
```

Comment:
Nit: Should we append UserWorkloads here, so its easier to debug like we have for the rest of the resources, like I said its a nit, but we create this same resources elsewhere.

Reply:
oh of course :+1: let me add this here, e2e flakiness seems to be friendly nowadays.